### PR TITLE
Descriptor objects

### PR DESCRIPTION
**Pull request by JonasPfeifer05**
Added basic structs and funcionallities to build up an ER-model

### CODE REVIEW

File: shellborn/src/relation/connection.rs [hunk 1]
```diff
@@ -0,0 +1,75 @@
+use crate::relation::cardinality::Cardinality;
+
+#[derive(PartialEq, Debug)]
+pub struct Connection {
+    pub to_name: String,
+    pub cardinalities: Cardinality,
+}
+
+impl Connection {
+    pub fn zero_to_one<S: Into<String>>(to_name: S) -> Self {
```
Comment (ToBinio) [inline shellborn/src/relation/connection.rs:10]: same as above having these functions shifts the job of parsing the connection from them self to the builder

**Review by ToBinio — CHANGES_REQUESTED** [review]
some comments.

think a lot is a bit overcomplicated and should be shifted to the parser

File: shellborn/src/attribute/mod.rs [hunk 2]
```diff
@@ -0,0 +1,115 @@
+use crate::attribute::attribute_decoration::AttributeDecoration;
+use crate::attribute::Attribute::{Concatenated, Single};
+
+pub mod attribute_decoration;
+pub mod builder;
+
+#[derive(Debug, PartialEq)]
+pub enum Attribute {
```
[the remark this answers is not in the document]
Comment (JonasPfeifer05) [reply shellborn/src/attribute/mod.rs:8]: yea seems resonable

File: shellborn/src/relation/builder.rs [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+
+impl RelationBuilder {
+    pub fn new<S: Into<String>>(name: S) -> Self {
+        Self {
+            name: name.into(),
+            connections: vec![],
+        }
+    }
+
+    pub fn add_zero_to_one<S: Into<String>>(mut self, to_name: S) -> Self {
```
[the remark this answers is not in the document]
Comment (JonasPfeifer05) [reply shellborn/src/relation/builder.rs:17]: It makes sense to add a type for custom values like 3 or so, yes
but same here as above

File: shellborn/src/relation/cardinality.rs [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+        Self {
+            from: FromCardinality::One,
+            to: ToCardinality::Many,
+        }
+    }
+}
+
+#[derive(PartialEq, Debug)]
+pub enum FromCardinality {
+    Zero,
```
[the remark this answers is not in the document]
Comment (JonasPfeifer05) [reply shellborn/src/relation/cardinality.rs:39]: It makes sense to add a type for custom values like 3 or so, yes
but same here as above

File: shellborn/src/relation/connection.rs (hunk 1 above)
Comment (JonasPfeifer05) [reply shellborn/src/relation/connection.rs:10]: same as above

File: shellborn/src/relation/cardinality.rs (hunk 4 above)
[the remark this answers is not in the document]
Comment (ToBinio) [reply shellborn/src/relation/cardinality.rs:39]: but doesn't that make parsing harder

File: shellborn/src/entity/builder.rs [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+            attributes: vec![],
+        }
+    }
+
+    pub fn set_weak(mut self, is_weak: bool) -> Self {
+        self.is_weak = is_weak;
+        self
+    }
+
+    pub fn add_basic<S: Into<String>>(mut self, name: S) -> Self {
```
[the remark this answers is not in the document]
Comment (JonasPfeifer05) [reply shellborn/src/entity/builder.rs]: removed! new generic function add_attribute

File: shellborn/src/relation/builder.rs (hunk 3 above)
[the remark this answers is not in the document]
Comment (JonasPfeifer05) [reply shellborn/src/relation/builder.rs]: removed! new generic function add_connection

File: shellborn/src/relation/cardinality.rs (hunk 4 above)
[the remark this answers is not in the document]
Comment (JonasPfeifer05) [reply shellborn/src/relation/cardinality.rs:39]: removed enums, changed to only use custom cardinalities

File: shellborn/src/relation/connection.rs (hunk 1 above)
Comment (JonasPfeifer05) [reply shellborn/src/relation/connection.rs:10]: removed enums, changed to only use custom cardinalities

**Review by ToBinio — APPROVED** [review]

**Pull request merged**